src/material-experimental/dialog/dialog-ref.ts

```diff
@@ -0,0 +1,137 @@
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+
+import {OverlayRef, GlobalPositionStrategy} from '@angular/cdk/overlay';
+import {Observable} from 'rxjs/Observable';
+import {map} from 'rxjs/operators/map';
+import {DialogPosition} from './dialog-config';
+import {CdkDialogContainer} from './dialog-container';
+
+/** Unique id for the created dialog. */
+let uniqueId = 0;
+
+/**
+ * Reference to a dialog opened via the Dialog service.
+ */
+export class DialogRef<T> {
+  /** The instance of the component in the dialog. */
+  componentInstance: T;
+
+  /** Whether the user is allowed to close the dialog. */
+  disableClose: boolean | undefined;
+
+  /** Result to be passed to afterClosed. */
+  private _result: any;
+
+  constructor(
+    protected _overlayRef: OverlayRef,
+    protected _containerInstance: CdkDialogContainer,
+    readonly id: string = `dialog-${uniqueId++}`) {
+
+    // If the dialog has a backdrop, handle clicks from the backdrop.
+    if (_containerInstance._config.hasBackdrop) {
+      _overlayRef.backdropClick().subscribe(() => {
+        if (!this.disableClose) {
+          this.close();
+        }
+      });
+    }
+
+    this.beforeClose().subscribe(() => {
+      this._overlayRef.detachBackdrop();
+    });
+
+    this.afterClosed().subscribe(() => {
+      this._overlayRef.detach();
+      this._overlayRef.dispose();
+      this.componentInstance = null!;
+    });
+  }
+
+  /** Gets an observable that emits when the overlay's backdrop has been clicked. */
+  backdropClick(): Observable<void> {
+    return this._overlayRef.backdropClick();
+  }
+
+  /**
+   * Close the dialog.
+   * @param dialogResult Optional result to return to the dialog opener.
+   */
+  close(dialogResult?: any): void {
+    this._result = dialogResult;
+    this._containerInstance._startExiting();
+  }
+
+  /**
+   * Updates the dialog's position.
+   * @param position New dialog position.
+   */
+  updatePosition(position?: DialogPosition): this {
+    let strategy = this._getPositionStrategy();
+
+    if (position && (position.left || position.right)) {
+      position.left ? strategy.left(position.left) : strategy.right(position.right);
+    } else {
+      strategy.centerHorizontally();
+    }
+
+    if (position && (position.top || position.bottom)) {
+      position.top ? strategy.top(position.top) : strategy.bottom(position.bottom);
+    } else {
+      strategy.centerVertically();
+    }
+
+    this._overlayRef.updatePosition();
+
+    return this;
+  }
+
+  /**
+   * Gets an observable that emits when keydown events are targeted on the overlay.
+   */
+  keydownEvents(): Observable<KeyboardEvent> {
+    return this._overlayRef.keydownEvents();
+  }
+
+  /**
+   * Updates the dialog's width and height.
+   * @param width New width of the dialog.
+   * @param height New height of the dialog.
+   */
+  updateSize(width: string = 'auto', height: string = 'auto'): this {
```

Comment:
What about min/max height/width?

Reply:
I think leaving them as `width` and `height` might be best since the config has `maxWidth` and `maxHeight` properties?

Reply:
You can't update the config once the dialog is open, though. The `OverlayConfig` has `updateSize` that takes an `OverlaySizeConfig` so that each dimension can be updated separately if necessary

Reply:
Gotcha, I was thinking you wanted to *replace* `width` and `height` with `max-width` and `max-height`.  Now I see what you are wanting to do.